### PR TITLE
[CDAP-941] register partitioned file sets in hive; unit tests for M/R and Explore

### PR DESCRIPTION
**Pull request by anew**
title says all.

### CODE REVIEW

**Comment by anew** [comment]
Build running in Bamboo here: http://builds.cask.co/browse/CDAP-DUT752

**Comment by anew** [comment]
build passed: http://builds.cask.co/browse/CDAP-DUT752

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java [hunk 1]
```diff
@@ -353,11 +354,57 @@ public void addPartition(HttpRequest request, HttpResponder responder,
...
+  @Path("/datasets/{dataset}/partitions")
+  public void addPartition(HttpRequest request, HttpResponder responder,
+                           @PathParam("dataset") final String datasetName) {
+    try {
+      Dataset dataset = instantiateDataset(datasetName, responder);
+      if (dataset == null) {
+        return; // response sent by instantiateDataset()
+      }
+      if (!(dataset instanceof PartitionedFileSet)) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "not a partitioned dataset.");
```
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: Do you need to return after sending the string?

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java [hunk 2]
```diff
@@ -353,11 +354,57 @@ public void addPartition(HttpRequest request, HttpResponder responder,
...
+
+      PartitionKey partitionKey;
+      try {
+        partitionKey = PartitionedFileSetArguments.getOutputPartitionKey(properties, partitioning);
+      } catch (Exception e) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "invalid partition key: " + e.getMessage());
+        return;
+      }
+      if (partitionKey == null) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "no partition key was given.");
```
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: Same here. Return after sending string?

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java [hunk 3]
```diff
@@ -372,6 +419,49 @@ public void dropPartition(HttpRequest request, HttpResponder responder,
...
+  @Path("/datasets/{dataset}/deletePartition")
+  public void dropPartition(HttpRequest request, HttpResponder responder,
+                            @PathParam("dataset") final String datasetName) {
+    try {
+      Dataset dataset = instantiateDataset(datasetName, responder);
+      if (dataset == null) {
+        return; // response sent by instantiateDataset()
+      }
+      if (!(dataset instanceof PartitionedFileSet)) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "not a partitioned dataset.");
```
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: return?

File: cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java [hunk 4]
```diff
@@ -372,6 +419,49 @@ public void dropPartition(HttpRequest request, HttpResponder responder,
...
+
+      PartitionKey partitionKey;
+      try {
+        partitionKey = PartitionedFileSetArguments.getOutputPartitionKey(properties, partitioning);
+      } catch (Exception e) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "invalid partition key: " + e.getMessage());
+        return;
+      }
+      if (partitionKey == null) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "no partition key was given.");
```
Comment (chtyim) [inline cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreExecutorHttpHandler.java]: return?

**Comment by anew** [comment]
@chtyim I addressed the comments you made so far. Anything else?

**Comment by chtyim** [comment]
Just couple comments. Rest looks good.

**Pull request merged**